org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationPatching.cs

```diff
@@ -0,0 +1,442 @@
+            Debug.WriteLine( string.Format( "Comparing {0} Top-Level Domains.", tlds.Length ) );
+
+            // ENews from DataView To List
+            var qry = enewsDV.GetQuery();
+            var eNewsData = qry.Select( item => item ).ToList();
+            HashSet<string> eNewsEmails = new HashSet<string>();
```

Comment:
We used a HashSet here because our use case was to check if a person is in the eNews data view quickly. We use the .contains() method a few lines below..

https://github.com/lakepointe/Rock-LPC/pull/12/files/1db4f85efd0a2ea4da4343e5b734838c089a5174#diff-71af37808a48c098da3cf5525d40257e4fa34ec71afa197fed2799a54df6cb56R305

List contains should be order n time, whereas hashset contains should be order 1

Reply:
I can't speak for other instances where HashSet was used, but I specifically asked Shane to implement it here.

Reply:
https://david.gardiner.net.au/2021/10/fastest-contains.html

Reply:
I tested a very similar scenario on a .NET Framework console app and got a consistent performance increase while using a `List` or `Array` over a `HashSet`. The difference may come from adding items to the iterable. `HashSet` has some additional overhead when adding items, which we are doing a lot of. I'd be happy to show you my test scenario in our dev meeting and talk about this.

Reply:
Actually, I just realized the `Contains` calls were inside some pretty big loops and getting called a lot more than I thought. `HashSet` is the right choice here. I'll resolve this conversation and delete the others.

Reply:
<img src="https://media4.giphy.com/media/oe33xf3B50fsc/giphy.gif"/>

Reply:
Jeff said our conflict wasn't explosive, so I decided to add an explosion here.